### PR TITLE
Validate `Authorization` header in Azure test fixture

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today the Azure test fixture accepts all requests, but we should be
checking that the `Authorization` header is at least present and
approximately correct. This commit adds support for this check.

### CODE REVIEW

**Review by mhl-b — APPROVED** [review]
lgtm

File: x-pack/plugin/repositories-metering-api/qa/azure/src/javaRestTest/java/org/elasticsearch/xpack/repositories/metering/azure/AzureRepositoriesMeteringIT.java [hunk 1]
```diff
@@ -30,7 +30,8 @@ public class AzureRepositoriesMeteringIT extends AbstractRepositoriesMeteringAPI
     private static AzureHttpFixture fixture = new AzureHttpFixture(
         USE_FIXTURE ? AzureHttpFixture.Protocol.HTTPS : AzureHttpFixture.Protocol.NONE,
         AZURE_TEST_ACCOUNT,
-        AZURE_TEST_CONTAINER
+        AZURE_TEST_CONTAINER,
+        AzureHttpFixture.startsWithPredicate("SharedKey " + AZURE_TEST_ACCOUNT + ":")
```
Comment (DiannaHohensee) [inline x-pack/plugin/repositories-metering-api/qa/azure/src/javaRestTest/java/org/elasticsearch/xpack/repositories/metering/azure/AzureRepositoriesMeteringIT.java:34]: where do the headers get generated that we're trying to match here by expecting "SharedKey ....:"?

**Review by DiannaHohensee — COMMENTED** [review]
Unfamiliar with this code, so I have some questions, mostly.

**Comment by DiannaHohensee** [comment]
Ah, looks like Mikhail took care of this before I finished. Cool 👍

**Comment by DaveCTurner** [comment]
Thanks @DiannaHohensee, you raise valid points, we can continue to iterate on them in https://github.com/elastic/elasticsearch/pull/111252.

**Pull request merged**